src/main/java/org/jenkinsci/plugins/workflow/support/steps/stash/StashStep.java

```diff
@@ -111,29 +140,53 @@ public boolean isAllowEmpty() {
             this.step = step;
         }
 
-        @Override protected Void run() throws Exception {
-            StashManager.stash(getContext().get(Run.class), step.name, getContext().get(FilePath.class), getContext().get(Launcher.class), getContext().get(EnvVars.class), getContext().get(TaskListener.class), step.includes, step.excludes,
+        private void verbose(List<File> files) throws Exception {
+            if (step.verbose) {
+                files.stream().forEach(file -> {
+                    PrintStream logger = null;
+                    try {
+                        String name = getContext().get(Run.class).getParent().getName();
+                        logger = getContext().get(TaskListener.class).getLogger();
+                        String gibberish = file.getCanonicalPath().split("/" + name + "/")[0];
+                        logger.println("stashed file " + StringUtils.substring(file.getCanonicalPath(), gibberish.length()));
+                    } catch (IOException | InterruptedException e) {
+                        logger.println("stashed file " + file.getName());
+                    }
+
+                });
+            }
+        }
+
+        @Override
+        protected Void run() throws Exception {
+            List<File> stash = StashManager.stash(getContext().get(Run.class), step.name, getContext().get(FilePath.class), getContext().get(Launcher.class), getContext().get(EnvVars.class), getContext().get(TaskListener.class), step.includes, step.excludes,
```

Comment:
`File`? Does not smell right. That could only refer to paths on the controller, and stashes are archived into a tarball. If these “files” are on an agent, then you cannot use `File` to refer to them. You _could_ use `FilePath` but (as above) you probably just meant to use `String` (relative paths).

Reply:
my goal is to pass something neutral, like a file (or FilePAth), or even a String. I did passed a File in order to provide more info in higher levels, not meaning to pass the actual file. Indeed, something like a DTO, which could be created.

Reply:
Just `String` should suffice. `stash` and `unstash` deal with paths relative to the contextual working dir, so the absolute file path is not really interesting.

Reply:
I would say an URI :)